### PR TITLE
Update and modify \mnote function and location

### PR DESCRIPTION
**Pull request by amoodie**
Let's discuss before you accept this PR. 

Initially I set out to just change the `\mnote` to be single spaced, because I think it makes more sense this way. My effort turned into something more (i.e., I went down the rabbit hole), and I moved the entire `\mnote` logic outside the `\iftoggle{@draft}` logic. 

I think this actually makes more sense to do, because it puts _all_ the code relating to `\mnote` in one place. Indeed, I think the same should be done for `\cnote` and `\pnote`. In this way, only the geometry, line spacing, and line numbering would be handled inside the `\iftoggle{@draft}`. This makes sense because they are really the `@draft` specific options. The `note`s should have their own arbitrary logic that can be modified by any other number of options (e.g,. `@nonotes`).

So, I think this PR represents the correct first step in modifying the code to be more maintainable and logical. We could either merge in this PR and then separately move `\cnote` and `\pnote`, or I can do the refactor for those features and merge them in with this PR. Let me know what you think, and how you want to proceed.

I demonstrate the new `\mnote` below using the following code:
```
\section{This is another section}
Is there a difference between this text and some nonsense like ``Huardest gefburn''? Kjift – not at all!  \cnote{this needs a ref}
A blind text like this gives you information about the selected font, how the letters are written and \pnote{what's this?} an impression \mnote{margin note can be used without highlighting} of the look.
This text should contain all letters of the alphabet and it should be written \mnote[in of itself of]{typo! check this stuff!} the original language. 
There is no need for special content, but the length of words should match the language. 
```
Note that `\mnote` now takes an _optional_ argument in square brackets. This is the text that is highlighted if given. Otherwise, the note is just put on the side with a line to the in-code location.

`draft` mode:
draft

`draft,nonotes` mode:
draft_nonotes

`final` mode:
final

Note, there are some weird spaces that remain in the `nonotes` and `final` modes, e.g., in front of `\cnote` call right after "not at all!". I'll open a separate issue about this, it is unrelated to the changes made herein.

### CODE REVIEW

**Comment by amoodie** [comment]
@ericbarefoot, just wanted to check in on the status of your review.

I'm sure it will be very thorough, considering you have been working on it for over 7 weeks!

File: compact_proposal/compact_proposal.cls
Comment (ericbarefoot) [inline compact_proposal/compact_proposal.cls:72]: Some fo the stuff like this I don't really understand yet, but I think I see how the general structure of this works. You'll have to teach me the specifics when you get back.

**Review by ericbarefoot — APPROVED** [review]
I like the idea here. I'll try and replicate this process with cnote and pnote myself to learn how it works, and then you should review before we merge those changes.

**Comment by ericbarefoot** [comment]
I like the direction you put here. I'll attempt the same thing with pnote and cnote later.

**Comment by amoodie** [comment]
@ericbarefoot great, glad you liked it. I will definitely be happy to show you how the bits I wrote work, though they rely heavily on the packages we are using for managing options etc.

**Pull request merged**
